workflows/prognostic_c48_run/runtime/segmented_run/run.py

```diff
@@ -22,8 +22,23 @@ def cwd(path):
 def find(path: str):
     return glob.glob(os.path.join(path, "**"), recursive=True)
 
+def compose_simulation_command(mpi_launcher: str, nprocs: int): 
 
-def run_segment(config: dict, rundir: str):
+    command = ["-n " + str(nprocs), sys.executable, "-m", "mpi4py", runfile.absolute().as_posix()]
+    
+    if mpi_launcher == "mpirun":
+        command.insert(0, mpi_launcher)
+
+    elif mpi_launcher == "srun":
+        command.insert(0, mpi_launcher)
+        command.insert(1, "--export=ALL")
+    else:
+        raise Exception("Unrecognized MPI launcher option. Please choose between 'mpirun' or 'srun'.")
```

Comment:
It would be good to stick with the literal name of the option here.  I think a [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError) also makes the most sense to raise, rather than a generic `Exception`:
```suggestion
        raise ValueError("Unrecognized mpi_launcher option. Please choose between 'mpirun' or 'srun'.")
```